File: CS/Controllers/CustomersController.cs
using DevExpress.Web.Mvc;
using DevExpress.Xpo;
using System.Collections.Generic;
using System.Linq;
using System.Web.Mvc;

namespace DevExpressMvcApplication.Controllers {
    public class CustomersController : BaseXpoController<Customer> {
        public ActionResult Index() {
            return View(GetCustomers());
        }

        public ActionResult IndexPartial() {
            return PartialView("IndexPartial", GetCustomers());
        }

        [HttpPost]
        public ActionResult EditCustomer([ModelBinder(typeof(DevExpressEditorsBinder))] CustomerViewModel customer) {
            Save(customer);
            return PartialView("IndexPartial", GetCustomers());
        }

        [HttpPost]
        public ActionResult DeleteCustomer([ModelBinder(typeof(DevExpressEditorsBinder))] CustomerViewModel customer) {
            Delete(customer);
            return PartialView("IndexPartial", GetCustomers());
        }

        IEnumerable<CustomerViewModel> GetCustomers() {
            return (from c in XpoSession.Query<Customer>().ToList()
                    select new CustomerViewModel() { ID = c.Oid, Name = c.Name }).ToList();
        }
    }
}

File: VB/Controllers/CustomersController.vb
Imports DevExpress.Web.Mvc
Imports DevExpress.Xpo
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web.Mvc

Namespace DevExpressMvcApplication.Controllers
	Public Class CustomersController
		Inherits BaseXpoController(Of Customer)

		Public Function Index() As ActionResult
			Return View(GetCustomers())
		End Function

		Public Function IndexPartial() As ActionResult
			Return PartialView("IndexPartial", GetCustomers())
		End Function

		<HttpPost>
		Public Function EditCustomer(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal customer As CustomerViewModel) As ActionResult
			Save(customer)
			Return PartialView("IndexPartial", GetCustomers())
		End Function

		<HttpPost>
		Public Function DeleteCustomer(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal customer As CustomerViewModel) As ActionResult
			Delete(customer)
			Return PartialView("IndexPartial", GetCustomers())
		End Function

		Private Function GetCustomers() As IEnumerable(Of CustomerViewModel)
			Return (
			    From c In XpoSession.Query(Of Customer)().ToList()
			    Select New CustomerViewModel() With {.ID = c.Oid, .Name = c.Name}).ToList()
		End Function
	End Class
End Namespace
